nornir/plugins/inventory/netbox.py

```diff
@@ -183,10 +243,29 @@ def __init__(
                     host["data"][cf] = value
                 host["data"].pop("custom_fields")
 
+            # Add secrets to host if it has one in Netbox
+            if nb_private_key:
+                # Filter secrets based on netbox secret role slug
+                f_secrets = [s for s in nb_secrets if s.get("role").get("slug") == nb_cred_role_slug]
+                # Add secrets to device if they exist
+                host_secret = [s for s in f_secrets if s.get("device").get("id") == dev.get("id")]
+                if host_secret:
+                    host_secret = host_secret[0]
+                    host["username"] = host_secret.get("name")
+                    host["password"] = host_secret.get("plaintext")
+
+            # management port from netbox config_context if it's available
+            if host.get("data").get("config_context").get("nornir").get("port"):
```

Comment:
I think this is something too specific to put into the plugin. It expects everyone to document the port to connect to be stored in a certain format, which is not default Netbox behavior.
Some other users might use for example a custom field on a device to set the port and would like to set the port based on that. 
There is no generic way to do this and in my opinion the plugin should only support behavior which is generic so that it works for everyone? Opinions?

Reply:
I agree, it felt dirty doing this. But if we just call it an opinionated netbox plugin, then we can do stuff like that :)

Seriously though, I was trying to make it such that a user can just swap out netbox inventory for simple inventory and it just work. I suppose the only other way to do this is to give the user another param to the module so they can specify a key name, that is "special, where they can put in nornir metadata like the port number, so it doesn't have to be hard-coded to `nornir` as it is in my example. Gives us flexibility for the future as well for other things, but for now, all that is needed is the port number.

Thoughts?